File: Introducción a la Programación Orientada a Objetos/Ejercicio-9/Python/Computadora.py
class Computadora:
    def __init__(self, marca, modelo, procesador, ramGB, almacenamientoGB):
        self.__marca = marca
        self.__modelo = modelo
        self.__procesador = procesador
        self.__ramGB = ramGB
        self.__almacenamientoGB = almacenamientoGB
        self.encendida = False

    def encender(self):
        if not self.encendida:
            self.encendida = True
            print("La computadora se está encendiendo...")
            print("¡Computadora encendida!")
        else:
            print("La computadora ya está encendida.")

    def apagar(self):
        if self.encendida:
            self.encendida = False
            print("La computadora se está apagando...")
            print("¡Computadora apagada!")
        else:
            print("La computadora ya está apagada.")

    def mostrar_estado(self):
        estado = "encendida" if self.encendida else "apagada"
        print(f"Estado actual: {estado}")

    # Getters (no son estrictamente necesarios en Python, pero los incluyo para equivalencia)
    def get_marca(self):
        return self.__marca

    def get_modelo(self):
        return self.__modelo

    def get_procesador(self):
        return self.__procesador

    def get_ramGB(self):
        return self.__ramGB

    def get_almacenamientoGB(self):
        return self.__almacenamientoGB

    def is_encendida(self):
        return self.encendida


if __name__ == "__main__":
    miPC = Computadora("Asus", "ExpertCenter AiO", "Intel ® Celeron ®", 8, 32)
    print("-----------------------------------------------------------------------")
    print("Computadora")
    print("-----------------------------------------------------------------------")
    print(f"- Marca: {miPC.get_marca()}")
    print(f"- Modelo: {miPC.get_modelo()}")
    print(f"- Procesador: {miPC.get_procesador()}")
    print(f"- RAM: {miPC.get_ramGB()} GB")
    print(f"- Almacenamiento: {miPC.get_almacenamientoGB()} GB")
    print("-----------------------------------------------------------------------")
    miPC.mostrar_estado()
    print("-----------------------------------------------------------------------")
    print("\nEncendiendo la computadora...")
    miPC.encender()
    miPC.mostrar_estado()
    print("-----------------------------------------------------------------------")
    print("\nApagando la computadora...")
    miPC.apagar()
    miPC.mostrar_estado()
    print("-----------------------------------------------------------------------")

File: Introducción a la Programación Orientada a Objetos/Ejercicio-9/Java/src/labmaquinas/Computadora.java
package labmaquinas;

public class Computadora {

	private String marca;

	private String modelo;

	private String procesador;

	private int ramGB;

	private int almacenamientoGB;

	private boolean encendida;
	

	public Computadora(String marca, String modelo, String procesador, int ramGB, int almacenamientoGB) {

	this.marca = marca;

	this.modelo = modelo;

	this.procesador = procesador;

	this.ramGB = ramGB;

	this.almacenamientoGB = almacenamientoGB;

	this.encendida = false; 

	}

	public void encender() {

	if (!encendida) {

	encendida = true;

	System.out.println("La computadora se está encendiendo...");
	System.out.println("¡Computadora encendida!");

	} else {

	System.out.println("La computadora ya está encendida.");

	}

	}

	public void apagar() {

	if (encendida) {

	encendida = false;

	System.out.println("La computadora  se está apagando...");
    System.out.println("¡Computadora apagada!");

	} else {

	System.out.println("La computadora ya está apagada.");

	}

	}

	public void mostrarEstado() {

	String estado = encendida ? "encendida" : "apagada";
	System.out.println("Estado actual: " + estado);

	}

	public String getMarca() { 
		return marca; 
		}

	public String getModelo() { 
		return modelo; 
		}

	public String getProcesador() { 
		return procesador; 
		}

	public int getRamGB() { 
		return ramGB; 
		}

	public int getAlmacenamientoGB() { 
		return almacenamientoGB; 
		}

	public boolean isEncendida() { 
		return encendida; 
		}


	public static void main(String[] args) {

	Computadora miPC = new Computadora("Asus", "ExpertCenter AiO", "Intel ® Celeron ®", 8,32);
    System.out.println("-----------------------------------------------------------------------");
    System.out.println("Computadora");
    System.out.println("-----------------------------------------------------------------------");
	System.out.println("- Marca: " + miPC.getMarca());
	System.out.println("- Modelo: " + miPC.getModelo());
	System.out.println("- Procesador: " + miPC.getProcesador());
	System.out.println("- RAM: " + miPC.getRamGB() + " GB");
	System.out.println("- Almacenamiento: " + miPC.getAlmacenamientoGB() + " GB");
    System.out.println("-----------------------------------------------------------------------");
	miPC.mostrarEstado();
    System.out.println("-----------------------------------------------------------------------");
	System.out.println("\nEncendiendo la computadora...");
	miPC.encender();
	miPC.mostrarEstado();
    System.out.println("-----------------------------------------------------------------------");
	System.out.println("\nApagando la computadora...");
	miPC.apagar();
	miPC.mostrarEstado();
    System.out.println("-----------------------------------------------------------------------");
	}
}
